source/lac/petsc_precondition.cc

```diff
@@ -488,32 +490,62 @@ namespace PETScWrappers
     AssertThrow (ierr == 0, ExcPETScError(ierr));
 
     if (additional_data.output_details)
-      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
+#if PETSC_VERSION_GT(3,7,0)
```

Comment:
If the API changes with 3.7.0 and you compile the code only with >3.7.0 it won't work with the 3.7.0 release. Can't you do
```
#if <=3.6.0
do old stuff
#else
do new stuff
#endif
```
?